ui/raidboss/data/05-shb/alliance/the_tower_at_paradigms_breach.ts

```diff
@@ -468,6 +477,144 @@ const triggerSet: TriggerSet<Data> = {
       suppressSeconds: 5,
       response: Responses.aoe(),
     },
+    // Deploy Armaments
+    //
+    // This attack has two variations and can be cast from two headings. The
+    // first variation causes two line AOEs to hit with the center being
+    // safe. The second variation causes a single line AOE to be cast in the
+    // middle, with the sides safe. It can either be cast horizontally or
+    // vertically over the arena.
+    //
+    // There are several skill IDs involved:
+    // 5C00: indicate start of a middle line attack. Always appears with one
+    //       5C02 cast
+    // 5C03: indicate start of a two side lines attack. Always appears with
+    //       2x 5C05 casts
+    // 5C01: indicates a single line attack comboing with the other boss.
+    //       Always appears simultaneously with the other bosses abilities
+    //       and a 6078 cast.
+    // 5C04: indicates a two side lines attack comboing with the other boss
+    //       Always appears simultaneously with the other bosses abilities
+    //       and 2x 6079 casts.
+    //
+    // To handle overlap, we have one trigger which captures all of the
+    // events into an array. Then, a delayed trigger processes all of the
+    // triggers and determines what to message players.
+    {
+      id: 'Paradigm Meng-Zi/Xun-Zi Deploy Armaments Collect',
+      type: 'StartsUsing',
+      netRegex: NetRegexes.startsUsing({ id: ['5C00', '5C01', '5C03', '5C04'] }),
+      run: (data, matches) => {
+        data.deployArmaments ??= [];
+        if (matches.id === '5C00' || matches.id === '5C01') {
+          if (matches.heading === '1.570772')
+            data.deployArmaments.push(DeployArmaments.VerticalMiddle);
+          else if (matches.heading === '-4.792213E-05')
+            data.deployArmaments.push(DeployArmaments.HorizontalMiddle);
+        } else if (matches.id === '5C03' || matches.id === '5C04') {
+          if (matches.heading === '1.570772')
+            data.deployArmaments.push(DeployArmaments.VerticalSides);
+          else if (matches.heading === '-4.792213E-05')
+            data.deployArmaments.push(DeployArmaments.HorizontalSides);
+        }
+      },
+    },
+    {
+      id: 'Paradigm Meng-Zi/Xun-Zi Deploy Armaments Trigger',
+      type: 'StartsUsing',
+      netRegex: NetRegexes.startsUsing({ id: ['5C00', '5C01', '5C03', '5C04'], capture: false }),
+      delaySeconds: 0.25,
```

Comment:
Specifically, we could do one of I think two things here:

1) just increase the delay to 3 or 3.05 or 3.1 seconds. This would collapse the 3-second staggered casts into one trigger, but at the cost of making it potentially not warn you about one section with enough time to move for the first AOE.

2) have it print immediately to move, but follow up with a "move here" warning for the 2nd AOE. To capture this, we could keep track of the time for each buff, and when we get a "using" skill, we could clear that attack from the list. This would allow printing something like "go middle!" followed by "Go N/S middle" or "Go E/W" by "Go Corners"

Basically the (2) idea is that we'd track casts until the AOE actually cleared. It's a bit more work, and I'm not sure how much benefit it has.

Reply:
I agree that (2) seems like the better approach if you wanted to do something like this.